packages/syntax-core/src/SelectList/SelectList.test.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable testing-library/no-await-sync-events */
+import { render, screen } from "@testing-library/react";
+import SelectList from "./SelectList";
+import { expect, vi } from "vitest";
+import userEvent from "@testing-library/user-event";
+
+const SelectDropdown = ({
+  numberOfOptions,
+  onChange,
+}: {
+  numberOfOptions: number;
+  onChange: () => void;
+}) => {
+  const options = [];
```

Comment:
@nathan-ong a more functional way to write this:

```js
(new Array(numberOfOptions)).fill(0).map((item, index) => ({
  value: `value-${index + 1}`,
  label: `label-${index + 1}`
}))
```

Also differentiates between a `label` and `values`

Reply:
Ahh very cool. Is there a need to fill with `0` since we're just using it to map over?

Reply:
I was looking into this a little more and came across this way of doing it 
```
Array.from({length:numberOfOptions},(item, index)=> ({value: index + 1, label: index + 1}))
```
Slightly more space optimized because we're not creating an additional array by mapping over the first array?